matlab_simulation/05-estimation/kalman_filter/motion_model.m

```diff
@@ -0,0 +1,65 @@
+function [Ad,Bd,R,n] = motion_model(example,dt)
+
+% Provides discretized Ad(state update), Bd(control) and R(disturbance) 
+% matrices for the motion model
+
+% Inputs:
+%       example   : Example under considertaion [1,2 or 3]
+%       dt        : time interval in seconds
+
+% Outputs:
+%       Ad        : State update
+%       Bd        : control
+%       R         : Disturbance
+%       n         : state dimension
+
+% Example 1: Temperature
+if example == 1
+    Ad = 0.8;
+    Bd = 3;
+    R = 2;
+    n = length(Ad(1,:));
```

Comment:
n = length(Ad(1,:)); can be taken out of the if loops so you do not have to repeat it 3 time, but this is very minor